pandas/io/json/_normalize.py

```diff
@@ -469,7 +469,16 @@ def _recursive_extract(data, path, seen_meta, level=0):
             for obj in data:
                 for val, key in zip(_meta, meta_keys):
                     if level + 1 == len(val):
-                        seen_meta[key] = _pull_field(obj, val[-1])
+                        try:
+                            seen_meta[key] = _pull_field(obj, val[-1])
+                        except KeyError as e:
```

Comment:
is it possible to move this logic into `_pull_field` itself?

Reply:
@jreback Yeah I can but `_pull_field` is also used to extract the records specified by `record_path`. So, if there's a try...except there, then the function would also be filtering out elements of `data` where the `record_path` doesn't exist (which is pretty different from the current behavior that raises a KeyError regardless of what `errors` is set to). Granted, that behavior could be useful since it prevents people from having to preprocess `data` if their `record_path` isn't guaranteed to be in each element. Thoughts?

Reply:
Actually, seems like that behavior was requested by a few people in [#21830](https://github.com/pandas-dev/pandas/issues/21830) as well, but yeah still not sure if it's desirable.